frontend/src/metabase/admin/settings/notifications/NotificationSettings.tsx

```diff
@@ -0,0 +1,134 @@
+import { useState } from "react";
+import { Link } from "react-router";
+import { t } from "ttag";
+
+import { useListChannelsQuery } from "metabase/api/channel";
+import {
+  Box,
+  Button,
+  Flex,
+  Icon,
+  type IconName,
+  Paper,
+  Stack,
+  Text,
+  Title,
+} from "metabase/ui";
+import type { NotificationChannel } from "metabase-types/api";
+
+import { CreateWebhookModal } from "./CreateWebhookModal";
+import { EditWebhookModal } from "./EditWebhookModal";
+
+type NotificationModals = null | "create" | "edit";
+
+export const NotificationSettings = () => {
+  const [webhookModal, setWebhookModal] = useState<NotificationModals>(null);
+  const [currentChannel, setCurrentChannel] = useState<NotificationChannel>();
+
+  const { data: channels } = useListChannelsQuery();
+
+  return (
+    <>
+      <Box w="47rem">
+        <Title mb="1.5rem">{t`Slack`}</Title>
+        <Link to="/admin/settings/notifications/slack">
+          <Paper shadow="0" withBorder p="lg" w="47rem" mb="2.5rem">
+            <Flex gap="0.5rem" align="center" mb="0.5rem">
+              <Icon name="slack_colorized" />
+              <Title order={2}>{t`Connect to slack`}</Title>
+            </Flex>
+            <Text>
+              {t`If your team uses slack, you can send dashboard subscriptions and
+            alerts there`}
+            </Text>
+          </Paper>
+        </Link>
+
+        <Flex justify="space-between" align="center" mb="1.5rem">
+          <Title>{t`Alerts webhook`}</Title>{" "}
+          {channels?.length !== 0 && (
+            <Button
+              variant="subtle"
+              compact
+              leftIcon={<Icon name="add" />}
+              onClick={() => setWebhookModal("create")}
+            >{t`Add another`}</Button>
+          )}
+        </Flex>
+        {channels?.length === 0 ? (
+          <ChannelBox
+            title={t`Add a webhook`}
+            description={t`Specify a webhook URL where you can send the content of Alerts`}
+            onClick={() => setWebhookModal("create")}
+            icon="webhook"
+          />
+        ) : (
+          <Stack>
+            {channels?.map(c => (
+              <ChannelBox
+                key={`channel-${c.id}`}
+                title={c.name}
+                description={c.description}
+                onClick={() => {
+                  setWebhookModal("edit");
+                  setCurrentChannel(c);
+                }}
+                icon="webhook"
+              />
+            ))}
+          </Stack>
```

Comment:
In certain cases there will be an empty `Stack` - will that look ok? I'm guessing yes

Reply:
it'll look fine, but the stack will only render if channels.length > 0

Reply:
@npfitz you could do:

```typescript
channels?.length
? <Stack>{channels.map... }</Stack>
: <ChannelBox... >
```
This groups together the case where `channels` is empty with the case where it is falsy. On lines 58-70 above it looks like the "Add a webhook" component is _not_ displayed if `channels` is falsy - is that desired?

Reply:
Rearranged it a bit so that we just have a Boolean that dictates if we have any channels. If there are no channels, we show a different button to add them.